### PR TITLE
ETH tests and push signatures

### PR DESCRIPTION
**Pull request by matiu**
Adds multiple tests for BWC and BWS for ETH transaction creation, signature, and broadcast.
Refactor signTx to push signatures to BWS.

### CODE REVIEW

File: packages/bitcore-wallet-service/src/lib/model/wallet.ts [hunk 1]
```diff
@@ -188,6 +188,10 @@ export class Wallet {
     return this.n > 1;
   }
 
+  isUTXOCoin() {
```
Comment (cmgustavo) [inline packages/bitcore-wallet-service/src/lib/model/wallet.ts]: Nice!

File: packages/bitcore-wallet-service/src/lib/server.ts [hunk 2]
```diff
@@ -2476,34 +2517,41 @@ export class WalletService {
   }
 
   _validateAddr(wallet, inaddr, opts) {
-    const A = Bitcore_[wallet.coin].Address;
+    if (wallet.coin == 'eth') {
+      try {
+        Validation.validateAddress(
+          wallet.coin.toUpperCase(),
+          'mainnet',
```
Comment (justinkook) [inline packages/bitcore-wallet-service/src/lib/server.ts]: can we do wallet.network? For ETH it results in the same derivation path for all networks.

File: packages/bitcore-wallet-client/src/lib/key.ts [hunk 3]
```diff
@@ -409,14 +409,14 @@ export class Key {
       const addressPath = Constants.PATHS.SINGLE_ADDRESS;
       const privKey = xpriv.deriveChild(addressPath).privateKey;
       const tx = t.uncheckedSerialize();
-      const signedRawTx = Transactions.sign({
+      const signature = Transactions.getSignature({
         chain: txp.coin.toUpperCase(),
         tx,
         key: { privKey: privKey.toString('hex') },
-        from: txp.from
+        from: txp.from,
```
Comment (justinkook) [inline packages/bitcore-wallet-client/src/lib/key.ts]: `from: txp.from,` is not used as a param in `getSignature` anymore

Comment (matiu) [reply packages/bitcore-wallet-client/src/lib/key.ts]: thanks, fixed.

**Comment by micahriggan** [comment]
There are quite a few tests failing. Should we fix those in a separate PR, or are they related to this one

File: packages/bitcore-wallet-service/src/lib/server.ts [hunk 4]
```diff
@@ -2520,8 +2520,8 @@ export class WalletService {
     if (wallet.coin == 'eth') {
       try {
         Validation.validateAddress(
-          wallet.coin.toUpperCase(),
-          wallet.network,
+          wallet.coin.tUpperCase(),
```
Comment (justinkook) [inline packages/bitcore-wallet-service/src/lib/server.ts]: typo `toUpperCase()`

**Pull request merged**